src/validators/mosaicId.validator.ts

```diff
@@ -27,3 +27,22 @@ export class MosaicIdValidator implements Validator<string> {
         }
     }
 }
+
+export class MosaicIdAliasValidator implements Validator<string> {
+    validate(value: string, context: ValidationContext): void {
+        const aliasTag = '@';
+        if (value.charAt(0) !== aliasTag) {
+            try {
+                const mosaic = new Mosaic(new MosaicId(value), UInt64.fromUint(0));
+            } catch (err) {
+                throw new ExpectedError('Introduce a mosaic id in hexadecimal format. Example: 941299B2B7E1291C');
+            }
+        } else {
+            try {
+                const mosaic = new Mosaic(new NamespaceId(value), UInt64.fromUint(0));
```

Comment:
You could check just ``try new NamespaceId(alias)`` instead of adding it inside a Mosaic object.